source/tests-integration/TestFixture.ts

```diff
@@ -92,6 +92,43 @@ export class TestFixture {
         return;
     }
 
+    public async takeBestOrder(marketAddress: string, type: BN, numShares: BN, price: BN, outcome: BN, tradeGroupID: string): Promise<void> {
+        const tradeContract = await this.contractDeployer.getContract("Trade");
+        const trade = new Trade(this.connector, this.accountManager, tradeContract.address, this.configuration.gasPrice);
+
+        let actualPrice = price;
+        if (type == new BN(1)) {
+            const market = new Market(this.connector, this.accountManager, marketAddress, this.configuration.gasPrice);
+            const numTicks = await market.getNumTicks_();
+            actualPrice = numTicks.sub(price);
+        }
+        const ethValue = numShares.mul(actualPrice);
+
+        const bestPriceAmount = await trade.publicTakeBestOrder_(type, marketAddress, outcome, numShares, price, tradeGroupID, { attachedEth: ethValue });
```

Comment:
Why does this function call both `publicTakeBestOrder_ ` and `publicTakeBestOrder`?

Reply:
to get the return value. At present the EVm only support return values from eth calls, not transactions